### PR TITLE
HPCC-28505 Remove null dereference

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - Please review. I have re-checked, and there are were no use cases that relied on a [deferenced] null return, all unconditionally use (and hence would have crashed if it were null) the returned reference.

File: dali/base/dadfs.cpp [hunk 1]
```diff
@@ -4093,9 +4093,9 @@ protected: friend class CDistributedFilePart;
 
     virtual IDistributedFilePart &queryPart(unsigned idx) override
     {
-        if (idx<parts.ordinality())
-            return queryParts().item(idx);
-        return *(IDistributedFilePart *)NULL;
+        if (idx>=parts.ordinality())
+            assertex(!"CDistributedFileBase::queryPart out of range");
```
Comment (ghalliday) [inline dali/base/dadfs.cpp:4097]: style: throwUnexpectedX("CDistributedFileBase::queryPart out of range") is cleaner/more modern.

**Review by ghalliday — APPROVED** [review]
@jakesmith looks good to merge.  Do you want to change the error code before I merge?

File: dali/base/dadfs.cpp (hunk 1 above)
Comment (ghalliday) [reply dali/base/dadfs.cpp:4097]: same in the function below

Comment (jakesmith) [reply dali/base/dadfs.cpp:4097]: yep, I'll change them.

Comment (jakesmith) [reply dali/base/dadfs.cpp]: @ghalliday - changed and squashed in.

**Pull request merged**